src/scripts/OSFramework/Helper/Device.ts

```diff
@@ -127,9 +127,16 @@ namespace OSFramework.Helper {
 		 * @memberof DeviceInfo
 		 */
 		private static _getUserAgent(userAgent = ''): string {
-			return userAgent.replace(' ', '') === ''
-				? window.navigator.userAgent.toLowerCase()
-				: userAgent.toLowerCase();
+			const cleanedUserAgent = userAgent.replace(' ', '');
+
+			if (cleanedUserAgent === '') {
+				if (sessionStorage.userAgent) {
+					return sessionStorage.userAgent.toLowerCase();
+				}
+				return window.navigator.userAgent.toLowerCase();
+			} else {
+				return userAgent.toLowerCase();
+			}
```

Comment:
I don't like the approach of spreading `return` in the code. Maybe some refactoring in the future would be good.

Reply:
Yup the code needs some refactoring.